core/src/consensus/tendermint/message.rs

```diff
@@ -133,6 +143,23 @@ impl ConsensusMessage {
             vote_step: VoteStep::new(height, view, step),
         }
     }
+
+    pub fn new_proposal(signature: Signature, header: &Header) -> Result<Self, ::rlp::DecoderError> {
+        let height = header.number();
+        let view = consensus_view(header)?;
+        Ok(ConsensusMessage {
+            signature,
+            block_hash: Some(header.hash()),
+            vote_step: VoteStep::new(height as Height, view, Step::Propose),
+        })
+    }
+
+    pub fn verify(&self) -> Result<Address, Error> {
```

Comment:
Returning `Address` from `verify` function looks weird. How about `verifier` or `signer`?

Reply:
OK. I'll use the `signer`.